Source/Tools/LinuxEmulation/LinuxSyscalls/SyscallsSMCTracking.cpp

```diff
@@ -312,6 +313,36 @@ uint64_t SyscallHandler::GuestMremap(bool Is64Bit, FEXCore::Core::InternalThread
   return Result;
 }
 
+int SyscallHandler::OpenCodeMapFile() {
+  // Query from FEXServer whether this is the first instance of this executable; if it is, also enable code dumping!
+  FEX_CONFIG_OPT(RootFSPath, ROOTFS);
+  FEX_CONFIG_OPT(Multiblock, MULTIBLOCK);
+  auto ProgramName = FEXCore::Config::Get(FEXCore::Config::CONFIG_APP_FILENAME);
+  LOGMAN_THROW_A_FMT(ProgramName && ProgramName.value()->c_str()[0] == '/', "");
+
+  // Check RootFS first, then the plain path
+  auto ProgramFD = open((RootFSPath() + ProgramName.value()->c_str()).c_str(), O_RDONLY);
+  if (ProgramFD == -1) {
+    ProgramFD = open(ProgramName.value()->c_str(), O_RDONLY);
+  }
+  if (ProgramFD == -1) {
+    return -1;
+  }
+
+  int CodeMapFD = FEXServerClient::RequestCodeMapFD(FEXServerClient::GetServerFD(), ProgramFD, Multiblock);
+  close(ProgramFD);
+  if (CodeMapFD == -1) {
+    return -1;
+  }
+
+  FM.SetProtectedCodeMapFD(CodeMapFD);
+
+  // Ensure the file descriptor is closed in fork children
```

Comment:
I've seen this comment a few times. `CLOEXEC` specifically automatically closes the FD on execve{at,}. It doesn't close on fork. In /most/ cases it's usually always fork+exec but doesn't need to be.